jvm/src/main/java/com/microsoft/bond/protocol/StreamHelper.java

```diff
@@ -0,0 +1,45 @@
+package com.microsoft.bond.protocol;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+/**
+ * Contains helper methods for working with input and output streams.
+ */
+final class StreamHelper {
+
+    /**
+     * Reads a single byte from the input, raising an EOFException if end of stream is reached.
+     * @param inputStream source input stream
+     * @return a byte
+     * @throws EOFException if end of stream reached
```

Comment:
Agree, but in this case the very point of the method is to raise that exception if end of stream is reached. Hence it was reasonable to add the comment.

---
In reply to: [116106337](https://github.com/tstein/bond/pull/4#discussion_r116106337) [](ancestors = 116106337)